code/game/objects/items/wayfinding.dm

```diff
@@ -1,121 +1,250 @@
+	var/ppt_cost = 0 //Jan 9 '21: 2560 had its difficulties for NT as well
 	var/expression_timer
+	///Avoid being Reddit.
+	var/funnyprob = 2
+	///List of slogans used by the dispenser to attract customers.
+	var/list/slogan_list = list("Find a wayfinding pinpointer? Give it to me! I'll make it worth your while. Please. Daddy needs his medicine.", //last sentence is a reference to Sealab 2021
+								"See a wayfinding pinpointer? Don't let it go to the crusher! Recycle it with me instead. I'll pay you!", //I see these things heading for disposals through cargo all the time
+								"Need the disk? Can't get a pinpointer? Buy a wayfinding pinpointer and find the captain's office today!",
+								"Bleeding to death? Can't read? Find your way to medbay today!", //there are signs that point to medbay but you need basic literacy to get the most out of them
+								"Voted tenth best pinpointer in the universe in 2560!", //there were no more than ten pinpointers in the game in 2020
+								"Helping assistants find the departments they tide since 2560.", //not really but it's advertising
+								"These pinpointers are flying out the airlock!", //because they're being thrown into space
+								"Grey pinpointers for the grey tide!", //I didn't pick the colour but it works
+								"Feeling lost? Find direction.",
+								"Automate your sense of direction. Buy a wayfinding pinpointer today!",
+								"Feed me a stray pinpointer.", //American Psycho reference
+								"We need a slogan!") //Liberal Crime Squad reference
+	///Number of the list entry of the slogan we're up to.
+	var/slogan_entry = 0
+	///Next tick we can say a slogan.
+	COOLDOWN_DECLARE(next_slogan_tick)
+	///Next tick the dispenser's spew rejection of non-wayfinding pinpointers can be triggered.
+	COOLDOWN_DECLARE(next_spew_tick)
 
 /obj/machinery/pinpointer_dispenser/Initialize(mapload)
 	. = ..()
-	var/datum/bank_account/civ_acc = SSeconomy.get_dep_account(ACCOUNT_CIV)
+	var/datum/bank_account/civ_acc = SSeconomy.get_dep_account(payment_department)
 	if(civ_acc)
 		synth_acc.transfer_money(civ_acc, start_bal) //float has to come from somewhere, right?
 
 	synth_acc.account_holder = name
 
-	desc += " Only [ppt_cost] credits! It also likes making costumes..."
+	desc += " [ppt_cost ? "Only [ppt_cost] credits! " : ""]It also synthesises costumes for some reason."
 
-	set_expression("neutral")
+	power_change()
+
+	COOLDOWN_START(src, next_slogan_tick, COOLDOWN_SLOGAN)
+	slogan_list = shuffle(slogan_list) //minimise repetition
+
+/obj/machinery/pinpointer_dispenser/power_change()
+	. = ..()
+	cut_overlays()
+	if(powered())
+		set_expression("veryhappy", 2 SECONDS) //v happy to be back in the pinpointer business
+		START_PROCESSING(SSmachines, src)
+
+/obj/machinery/pinpointer_dispenser/update_icon_state()
+	if(machine_stat & BROKEN)
+		set_light(0)
+	else if(powered())
+		set_light(1.4)
+	else
+		set_light(0)
```

Comment:
This doesn't seem to be the right proc for it.

Reply:
It works. What do you mean by wrong?

Reply:
`update_icon_state()` is the proc, and this does not touch upon `icon_state`

Reply:
Vending machines do it this way. update_icon_state() is called when a machine's power updates.

Reply:
that is rather unfortunate, but alright